numpy/core/src/umath/loops.c.src

```diff
@@ -138,38 +138,64 @@
         tout * out = (tout *)op1; \
         op; \
     }
+/*
+ * unfortunately gcc 6/7 regressed and we need to give it additional hints to
+ * vectorize inplace operations (PR80198)
+ * must only be used after op1 == ip1 or ip2 has been checked
+ * TODO: using ivdep might allow other compilers to vectorize too
```

Comment:
if someone cares about other compilers, you can likely use this type of pragma there too. Last I checked clang could not vectorize this code, with the pragma it might be able to but it needs testing.